pkg/pgmodel/migrate.go

```diff
@@ -58,101 +71,251 @@ func (t *mySrc) replaceSchemaNames(r io.ReadCloser) (io.ReadCloser, error) {
 	s = strings.ReplaceAll(s, "SCHEMA_DATA", dataSchema)
 	s = strings.ReplaceAll(s, "SCHEMA_DATA_SERIES", dataSeriesSchema)
 	s = strings.ReplaceAll(s, "SCHEMA_INFO", infoSchema)
-	r = ioutil.NopCloser(strings.NewReader(s))
-	return r, err
+	return s, err
 }
 
-func (t *mySrc) ReadUp(version uint) (r io.ReadCloser, identifier string, err error) {
-	r, identifier, err = t.Driver.ReadUp(version)
+func metadataUpdate(db *pgxpool.Pool, withExtension bool, key string, value string) {
+	/* Ignore error if it doesn't work */
+	if withExtension {
+		_, _ = db.Exec(context.Background(), fmt.Sprintf(metadataUpdateWithExtension, extSchema), key, value, true)
+	} else {
+		_, _ = db.Exec(context.Background(), metadataUpdateNoExtension, key, value, true)
+	}
+}
+
+func ensureVersionTable(db *pgxpool.Pool) error {
+	_, err := db.Exec(context.Background(), fmt.Sprintf(createMigrationsTable, migrationsTableName))
 	if err != nil {
-		return
+		return fmt.Errorf("error creating migration table: %w", err)
 	}
-	r, err = t.replaceSchemaNames(r)
-	return
+
+	return nil
 }
 
-func (t *mySrc) ReadDown(version uint) (r io.ReadCloser, identifier string, err error) {
-	r, identifier, err = t.Driver.ReadDown(version)
+func getDBVersion(db *pgxpool.Pool) (semver.Version, error) {
+	var version semver.Version
+	res, err := db.Query(context.Background(), fmt.Sprintf(getVersion, migrationsTableName))
+
 	if err != nil {
-		return
+		return version, err
+	}
+
+	for res.Next() {
+		err = res.Scan(&version)
 	}
-	r, err = t.replaceSchemaNames(r)
-	return
+
+	if err != nil {
+		return version, err
+	}
+
+	return version, nil
 }
 
-func metadataUpdate(db *sql.DB, withExtension bool, key string, value string) {
-	/* Ignore error if it doesn't work */
-	if withExtension {
-		_, _ = db.Exec(metadataUpdateWithExtension, key, value, true)
-	} else {
-		_, _ = db.Exec(metadataUpdateNoExtension, key, value, true)
+func setDBVersion(tx pgx.Tx, version *semver.Version) error {
+	if _, err := tx.Exec(context.Background(), fmt.Sprintf(truncateMigrationsTable, migrationsTableName)); err != nil {
+		return fmt.Errorf("unable to truncate migrations table: %w", err)
 	}
+
+	if _, err := tx.Exec(context.Background(), fmt.Sprintf(setVersion, migrationsTableName), version.String()); err != nil {
+		return fmt.Errorf("unable to set version in DB: %w", err)
+	}
+
+	return nil
 }
 
 var migrateMutex = &sync.Mutex{}
 
-// Migrate performs a database migration to the latest version
-func Migrate(db *sql.DB, versionInfo VersionInfo) (err error) {
-	migrateMutex.Lock()
-	defer migrateMutex.Unlock()
-	// The migration table will be put in the public schema not in any of our schema because we never want to drop it and
-	// our scripts and our last down script drops our shemas
-	driver, err := postgres.WithInstance(db, &postgres.Config{MigrationsTable: "prom_schema_migrations"})
+func execMigrationFiles(tx pgx.Tx, name string) error {
```

Comment:
Does this execute _all_ the migration files? if so it should be named such. (If not, it could still use a more meaningful name; I'm still not entirely sure what this function does, and how it differs from `installVersionScripts`)

Reply:
If, as I'm guessing now this is for non-versioned files, like the pre-init stuff that's only ever done once, or the idempotent things which are done each time, than `execNonVersionedMigrations` might work, though `name` probably can use a better name.